addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -261,13 +393,185 @@ odoo.define('website_slides.quiz', function (require) {
+            var $lastQuestion = this.$('.o_wslides_js_lesson_quiz_question:last');
+            if ($lastQuestion.length !== 0) {
+                $lastQuestion.after(event.data.view);
+            } else {
+                this.$el.prepend(event.data.view);
+            }
+            this.quiz.questionsCount++;
+            if (event.data.save_and_new) {
+                this._onCreateQuizClick();
+            } else {
+                this.$('.o_wsildes_quiz_question_input').empty();
+                this.$('.o_wslides_js_quiz_add_question').removeClass('d-none');
+            }
+        },
+
+        /**
+         * Display the updated question in place of the inputs.
+         * @param event
+         * @private
+         */
+        _displayUpdatedQuestion: function (event) {
+            event.data.$updatedQuestionEl.replaceWith(event.data.view);
+        },
+
+        /**
+         * If the user cancels the creation or update of a Question it resets the display
+         * of the updated Question or it displays back the buttons.
+         *
+         * @param reset
+         * @private
+         */
+        _resetDisplay: function (reset) {
+            if (reset.data.update) {
+                reset.data.$elem.html(reset.data.$oldElem);
+            } else {
+                this.$('.o_wsildes_quiz_question_input').empty();
+                if (this.quiz.questionsCount > 0) {
+                    this.$('.o_wslides_js_quiz_add_question').removeClass('d-none');
+                } else {
+                    this.$('.o_wslides_js_quiz_add_quiz').removeClass('d-none');
+                }
+            }
+        },
+
+        /**
+         * After deletion of a Question the display is refreshed with the removal of the Question
+         * the reordering of all the remaining Questions and the change of the new Question sequence
+         * if the QuestionFormWidget is initialized.
+         *
+         * @param event
+         * @private
+         */
+        _deleteQuestion: function(event) {
+            var questionId = event.data.questionId;
+            this.$('.o_wslides_js_lesson_quiz_question[data-question-id=' + questionId + ']').remove();
+            this.quiz.questionsCount--;
+            this._reorderQuestions();
+            var $newQuestion = this.$('.o_wslides_js_lesson_quiz_new_question .o_wslides_quiz_question_sequence');
+            $newQuestion.text(parseInt($newQuestion.text()) - 1);
+            if (this.quiz.questionsCount === 0) {
+                this.$('.o_wslides_js_quiz_add_quiz').removeClass('d-none');
+                this.$('.o_wslides_js_quiz_add_question').addClass('d-none');
+                this.$('.o_wslides_js_lesson_quiz_validation').addClass('d-none');
+            }
+        },
+    });
+
+    /**
+     * Dialog box shown when clicking the deletion button on a Question.
+     * When confirming it sends a RPC request to delete the Question.
+     */
+    var ConfirmationDialog = Dialog.extend({
```

Comment:
You should return all the Components created within this file at the end.

Something like: 

```
return {
    Quiz: Quiz,
    ConfirmationDialog: ConfirmationDialog,
    websiteSlidesQuizNoFullscreen: publicWidget.registry.websiteSlidesQuizNoFullscreen
}
```

You will have to adapt calls that `require('website_slides.quiz')` to:
`var Quiz = require('website_slides.quiz').Quiz`

That will allow other modules to override any logic added by this file.
Right now, if you want "website_sale_slides" to override your confirmation window logic, it can't.

Thanks!

Reply:
Ok, done !